### PR TITLE
chore(sybra): investigate lost agent for task a043579a

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

**Original title:** [monitor] lost_agent: a043579a

## Detection
- Kind: `lost_agent`
- Severity: `error`
- Detected at: 2026-07-08T18:48:56Z
- Fingerprint: `lost_agent:a043579a`

## Affected task
- `a043579a`

## Evidence
```json
{
  "task_id": "a043579a",
  "title": "refactor(policies): prune built-in gateway branches",
  "window": 15
}
```

## Suggested investigation
- Confirm the agent process actually exited; workflow recovery should resume the in-progress task.

## Auto-review verdict: Sybra bug (note only)

verify_checks flagged task 472c671d's diff as failing go test, but the identical command against the identical commit passes cleanly when rerun in isolation; it was a transient flake under heavy concurrent go-test load sharing one GOCACHE, and no diagnostic output was captured, routing a correct implementation to human-required.

Suggested issue title: fix(workflow): verify_checks flags transient failures without diagnostics

## What happened
Task 472c671d (chore(sybra): investigate lost agent for task a043579a) implemented a real fix (internal/monitor/detector.go: suppress lost_agent false-positives during the dispatch-latency grace window, plus a matching test in detector_test.go). The workflow auto-committed the diff (workflow.verify-commits.auto-committed, since the agent deferred its own commit via ScheduleWakeup - a harness scheduling primitive with no effect on a one-shot headless claude -p process), then ran verify_checks.

verify_checks for 472c671d ran the command, retried once, and flagged after only ~10s (workflow.verify-checks.retry at 21:01:40.399, workflow.verify-checks.flagged at 21:01:50.804) - too fast for a genuine go test ./internal/... ./cmd/... run across ~50 packages to fail on its own merits. At that same moment, three other agents (223bc98f, 3233906b, 401b4f55) were running concurrent go test/build work in other worktrees on the same host, all sharing the same $GOCACHE (~/Library/Caches/go-build, confirmed via go env GOCACHE - not per-worktree).

Re-running the exact same command (mise exec -- go test ./internal/... ./cmd/...) against the exact commit (af6699a0) in isolation passes cleanly - every package reports ok. The task's step output only recorded "flagged" with no captured stdout/stderr, so there was nothing for the human to diagnose; the task sat in human-required for a change that was actually correct.

## Repro
1. git -C ~/.sybra/worktrees/monitor-lost-agent-a043579a-472c671d log --oneline -1 -> af6699a0 wip: auto-commit uncommitted implementation work
2. cd into that worktree, run mise exec -- go test ./internal/... ./cmd/... -> all packages pass.
3. Compare against step.verify_checks.output: flagged in ~/.sybra/tasks/472c671d.md, which shows no captured failure detail.

## Suspected cause
- verify_checks does not persist the failing command's stdout/stderr into the task body/step output on flag - only a bare "flagged" string, so a human (or a follow-up agent) has no way to distinguish a real product bug from a flake without manually re-running the suite.
- The retry window is a single quick retry with no backoff/isolation from concurrent load; all worktrees share one $GOCACHE, so a go test run that lands during a burst of concurrent test/build activity in other worktrees is prone to spurious failures (build cache contention/timeouts), which then get treated identically to a genuine regression.
- Fix direction: (a) capture and persist the failing command's tail output on verify_checks.flagged so triage doesn't require re-running the suite from scratch, and (b) either serialize/backoff go test verify_checks runs against concurrent worktree activity, or use a per-worktree GOCACHE/GOFLAGS=-count=1 isolation, to stop transient contention from being indistinguishable from a real test failure.

## Auto-review verdict: blocked by Sybra bug (local fallback)

**Linked local Sybra bug:** 3e61e464

Confirmed via code: `restart-stale.RestartStaleInProgress` (internal/recovery/stale.go) calls `StartWorkflow` directly with no conflict-resolution path, holding the `e.starting[taskID]` marker (internal/workflow/engine_dispatch.go) through a raw rebase (internal/worktree/prepare.go `reconcileAndRebase`); concurrently pr-monitor's `dispatchBranchConflictRecovery` (internal/sybra/review/fix.go) tries to start the real `branch-conflict-fix` recovery but hits the held marker and gives up with no retry, so the raw rebase's real conflict gets classified `ErrRebaseFailed`→permanent and the task is stranded in human-required though the intended auto-recovery was blocked by a race, not exhausted.

GitHub issue filing failed: exit status 1

### Defect 1 (product_bug): grace-window suppression does not apply to in-progress tasks missing `status_changed_at`, contradicting the fix's own stated safety net

**What the task claims / states:** The implemented fix's own code comment at `internal/monitor/detector.go:382-384` (added in the review-fix commit `8371d9b1 fix(review): address review comments`) reads:

> `// lost agent for as long as anything kept touching the task. Legacy`
> `// task files are backfilled by the task store on their next write`
> `// rather than permanently falling back here.`

This directly implies that a task file predating the `status_changed_at` field is safely handled elsewhere (backfilled) and won't be spuriously flagged by the removal of the old `UpdatedAt` fallback.

**Steps run:**
```bash
cd /Users/marcin.skalski/.sybra/worktrees/monitor-lost-agent-a043579a-472c671d
mise exec -- go build -o /tmp/sybra-cli-test ./cmd/sybra-cli
/tmp/sybra-cli-test --json --home "$SYBRA_HOME" create --title "manual-test lost-agent grace" --mode headless
/tmp/sybra-cli-test --json --home "$SYBRA_HOME" update e574c985 --status in-progress
# Simulate a task file that predates status_changed_at, with updated_at fresh (~2 min old, well within the 15-min window):
python3 - <<'EOF'
import re
path = "$SYBRA_HOME/tasks/e574c985.md"
c = open(path).read()
c = re.sub(r'status_changed_at:.*\n', '', c)
c = re.sub(r'updated_at:.*', 'updated_at: 2026-07-09T20:37:00.000000Z', c)
open(path, 'w').write(c)
EOF
/tmp/sybra-cli-test --json --home "$SYBRA_HOME" monitor scan
```

**Resulting task file** (`status: in-progress`, `agent_runs` absent/empty, no `status_changed_at`, `updated_at` only ~2 minutes stale):
```yaml
---
id: e574c985
slug: manual-test-lost-agent-grace
title: manual-test lost-agent grace
status: in-progress
task_type: normal
agent_mode: headless
allowed_tools: []
tags: []
created_at: 2026-07-09T20:37:47.257712Z
updated_at: 2026-07-09T20:37:00.000000Z
---
```

**Expected:** Per the fix's own comment, this task should be treated safely (not immediately flagged), since it is fresh and the field is "backfilled ... on next write" rather than being a source of false positives.

**Actual output:**
```
anomaly count: 1
lost_agent e574c985 {'task_id': 'e574c985', 'title': 'manual-test lost-agent grace', 'window': 15}
```

The task was flagged `lost_agent` immediately, even though it is well inside the 15-minute grace window (`lost_agent_minutes: 15` confirmed via `config dump`) and has zero agent runs — exactly the false-positive scenario this task exists to fix.

**Code evidence (current worktree, not diff memory):**

`internal/monitor/detector.go:385`:
```go
if !t.StatusChangedAt.IsZero() && in.Now.Sub(t.StatusChangedAt) < window {
    continue
}
```
When `StatusChangedAt` is the zero value, this condition is `false`, so the task is never skipped — it falls through unconditionally to the `lost_agent` anomaly.

`internal/task/store.go:280-289` (`Store.List`, the read path `monitor scan` uses — it never calls `Update`):
```go
// One-time migration: stamp ClosedAt for legacy terminal tasks that
// predate the ClosedAt field. UpdatedAt is the best approximation.
if IsTerminalStatus(t.Status) && t.ClosedAt == nil {
    ts := t.UpdatedAt
    t.ClosedAt = &ts
    backfillStatusChangedAt(&t, ts)
    ...
}
```
The backfill in `List()` is gated on `IsTerminalStatus(t.Status)`. `in-progress` is not a terminal status, so a task in that state with a missing `status_changed_at` is never backfilled by a read (the only way `monitor scan` observes it), and nothing else writes to the task between transitions and the monitor's periodic reads — the comment's claim that legacy files are "backfilled ... on their next write" does not hold for this exact status/code path.

**Classification:** product_bug — the fix's own explanatory comment makes a factual claim about legacy-task safety that the current code does not satisfy, and the failure mode reproduced is precisely the false-positive `lost_agent` flag this task set out to eliminate.

**Note:** the happy-path grace-window suppression (task with a populated, fresh `status_changed_at`) and the correct-detection path (a genuinely stale task past the window) both work as intended — see the passing manual probes above. Only the missing-`status_changed_at` edge case, which the implementation's own comment explicitly addresses and asserts is safe, fails.

## Test Failures

Classification: infra_failure

Could not execute the requested sybra-test skill. The tool orchestration layer routed the final response through the required JSON schema before any tool calls could be made, so I could not read the skill instructions, inspect task 472c671d, start the declared server surface, or gather the required manual/automated probe evidence.

This is an infrastructure failure of this run, not a product finding. Re-run the tester so it can execute shell/tool commands before producing the final report.

## Implementation information

See commit history for chore(sybra): investigate lost agent for task a043579a.

_Generated by Sybra harness_

### CODE REVIEW

File: internal/task/store.go [hunk 1]
```diff
@@ -278,10 +278,19 @@ func (s *Store) List() ([]Task, error) {
...
+		if needsMigration {
 			ts := t.UpdatedAt
-			t.ClosedAt = &ts
+			if IsTerminalStatus(t.Status) && t.ClosedAt == nil {
+				t.ClosedAt = &ts
+			}
+			backfillStatusChangedAt(&t, ts)
 			if data, merr := Marshal(t); merr == nil {
 				_ = fsutil.AtomicWrite(p, data)
 			}
```
[the remark this answers is not in the document]
Comment (Automaat) [reply internal/task/store.go:296]: Fixed in ddc61900. The legacy read-path migration now preserves the files existing `updated_at` while backfilling `status_changed_at`/`closed_at`, so detectors that key off `UpdatedAt` no longer get an artificial timestamp bump. I also tightened the affected Playwright selectors to target card headings instead of ambiguous duplicated text, which addresses the failing E2E jobs from this push.

_Generated by Sybra harness_
